### PR TITLE
Use system specific newline encoding when doing IO (close #219)

### LINKED ISSUE #219: Use system specific newline encoding when doing IO
As a side effect of https://github.com/sol/hpack/pull/143 we lost system specific newline encoding.

### PR DESCRIPTION
**Pull request by sol**

### CODE REVIEW

**Comment by sol** [comment]
@snoyberg @tfausak any input on this?

File: src/Hpack/Utf8.hs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+readFile = fmap decodeText . B.readFile
+
+writeFile :: FilePath -> String -> IO ()
+writeFile name xs = withFile name WriteMode (`hPutStr` xs)
+
+putStr :: String -> IO ()
+putStr = hPutStr stdout
+
+hPutStrLn :: Handle -> String -> IO ()
+hPutStrLn h xs = hPutStr h xs >> hPutStr h "\n"
```
Comment (tfausak) [inline src/Hpack/Utf8.hs]: This should write `"\r\n"` when appropriate, right?

Comment (sol) [reply src/Hpack/Utf8.hs]: @tfausak `hPutStr` will take care of converting `"\n"` to "\r\n" if necessary.

Comment (tfausak) [reply src/Hpack/Utf8.hs]: Oops, right!

**Comment by sol** [comment]
Thanks for the feedback guys!

**Pull request merged**